assets/js/modules/analytics-4/datastore/api.test.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * `modules/analytics-4` data store: api tests.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import { STORE_NAME } from './constants';
+import { createTestRegistry, unsubscribeFromAll } from 'tests/js/utils';
+
+describe( 'modules/analytics-4 properties', () => {
+	let registry;
+
+	beforeAll( () => {
+		API.setUsingCache( false );
+	} );
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		// Receive empty settings to prevent unexpected fetch by resolver.
+		registry.dispatch( STORE_NAME ).receiveGetSettings( {} );
+	} );
+
+	afterAll( () => {
+		API.setUsingCache( true );
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'isAdminAPIWorking', () => {
+			test.each`
+			property   | webData    | errorProperty | errorWebData | expected
+			${ true }  | ${ true }  |  ${ false }   | ${ false }   | ${ true }
+			${ true }  | ${ false } |  ${ false }   | ${ false }   | ${ undefined }
+			${ false } | ${ true }  |  ${ false }   | ${ false }   | ${ undefined }
+			${ false } | ${ false } |  ${ false }   | ${ false }   | ${ undefined }
+			${ true }  | ${ true }  |  ${ true }    | ${ false }   | ${ false }
+			${ true }  | ${ true }  |  ${ false }   | ${ true }    | ${ false }
+			${ true }  | ${ true }  |  ${ true }    | ${ true }    | ${ false }
+			${ false } | ${ true }  |  ${ true }    | ${ true }    | ${ false }
+			${ true }  | ${ false } |  ${ true }    | ${ true }    | ${ false }
+			${ false } | ${ false } |  ${ true }    | ${ true }    | ${ false }
```

Comment:
Previously I had more descriptive individual tests (see commit 
https://github.com/google/site-kit-wp/pull/3235/commits/33d24a36b90faa04365ea0bfb82a9ffdb4ee0873 )

But the AC implies a lot of combinations so this test is more comprehensive, but less readable!

Reply:
I'm usually a fan of tests that are more verbose in exchange for readability—but this formatting is actually pretty handy and readable for what this is. Nice 👍🏻